elastic/kibana
# Remove feature flag for APM Alert Details page

## Summary

This PR removes the `xpack.observability.unsafe.alertDetails.apm.enabled` feature flag to enable the Alert Detail page for the APM Latency Threshold Rule. I've also removed the`alertDetailsAppSection` options along with the `alertDetailsUrl` action context variable for the remaining APM rules since the APM Alert Detail page should only be available for the APM Latency Threshold rule. 

I also changed`isAlertDetailsEnabledPerApp()` method to allow the `apm.transaction_duration` rule type to work but not  `apm.error_rate`, `apm.transaction_error_rate`, or `apm.anomaly` rule types. This change was necessary because the granularity of the feature flag was set to the solution level and didn't allow to set the flag per rule type.

### Testing

To test this PR you will need to have some APM data in your cluster.
1. Create an APM Latency Threshold Rule with a low threshold so it will trigger
2. Create an APM Error Count Rule with a low threshold so it will trigger
3. Click on the "Action Menu" (the ellipses) in the alert (in the Alert Table) for the latency threshold alert
4. Click on the "View alert details" menu item for the latency threshold alert
5. Verify it takes you to the new "alert details" page for APM
6. Repeat steps 3 & 4 for the error count alert BUT it should open the "alert flyout" instead.
